### ISSUE TITLE
Issue with add_to_errors and in_errors_if

### ISSUE THREAD

**Issue by qurgh** [formr/Formr#16]
I've come across an issue with add_to_errors and in_errors_if.

In my code, I'm doing my own validation process. When I find a field that hasn't validated properly, I use add_to_errors to put it in the errors array. After I add all the failed fields I have the system doing a var_dump on $form->errors() to make sure they are in there, and it correctly returns the list. 

In my form, I'm using in_errors_if to print out a message if the field is in the errors array. However, in_errors_if never returns anything. For example I have the following:
```
$output .= $form->input_text('lname', "Last Name:");
$output .= $form->in_errors_if('lname', 'Please enter the student\'s last name');
$output .= $form->input_date('dob', "Date of Birth:");
$output .= $form->in_errors_if('dob', 'Please enter the student\'s date of birth');
```
But even when lname and dob are both added using add_to_errors, the in_errors_if never triggers. 

I also tried using if ($form->in_errors('dob')) { blah } but that never returns true either. 

Am I missing something or is this a bug?

**Comment by qurgh** [comment]
I did some digging into the functions, and found what I think is an error in the in_errors function. The line reads:

`if (array_key_exists($key, $this->errors)) {`

But when I do a var_dump on $this->errors I see all the error fields are listed as values, not keys. The keys are just numbers. So this function always returns false. I changed it to 

`if (in_array($key, $this->errors)) {`

and everything started working as expected.
